File: 3-2.py
import pandas as pd

df = pd.read_csv('pandas\edu_bigdata_imp1.csv', encoding = 'big5', low_memory = False)
df_filtered = df[df['PseudoID'] == 71]
df_filtered = df_filtered[['dp001_review_start_time', 'dp001_review_end_time']].dropna()
df_filtered['dp001_review_start_time'] = pd.to_datetime(df_filtered['dp001_review_start_time'])
df_filtered['dp001_review_start_time'] = df_filtered['dp001_review_start_time'].dt.date
df_filtered['dp001_review_end_time'] = pd.to_datetime(df_filtered['dp001_review_end_time'])
df_filtered['dp001_review_end_time'] = df_filtered['dp001_review_end_time'].dt.date
unique_value = df_filtered[['dp001_review_start_time']].drop_duplicates()
print(unique_value)

File: 3-2.php
<?php

$filename = 'pandas/edu_bigdata_imp1.csv';

$csvData = array_map('str_getcsv', file($filename));
$header = array_shift($csvData);

$pseudoIDIndex = array_search('PseudoID', $header);
$startTimeIndex = array_search('dp001_review_start_time', $header);
$endTimeIndex = array_search('dp001_review_end_time', $header);

$filteredValues = [];

foreach ($csvData as $row) {
    if ($row[$pseudoIDIndex] == 71 && isset($row[$startTimeIndex]) && isset($row[$endTimeIndex])) {
        $filteredValues[] = [$row[$startTimeIndex], $row[$endTimeIndex]];
    }
}

$filteredValues = array_map("unserialize", array_unique(array_map("serialize", $filteredValues)));

foreach ($filteredValues as $row) {
    echo implode(', ', $row) . PHP_EOL;
}

?>
